HelpSource/Classes/Object.schelp

```diff
@@ -200,6 +200,113 @@ A String containing the resulting file's path.
 
 subsection::Equality and Identity
 
+Equally asks if the value of two objects is equal.
+Identity asks if they are the same object.
+There are a few classes of object where only one instance exists and is shared between all uses (a singleton).
+These objects include Symbols and numbers.
+
+Equality, tested for with code::==:: or code::!=:: is a relation between objects.
+So code::a == b:: is true when the object in the variable a and the object in b are equivalent in some essential aspect.
+Two strings, for example, are equal when they have the same characters in the same order.
+Identity, by contrast, is a relation that every object has to itself. So code::a === b:: is true only when the the variables a and b both contain the same object.
+
+code::
+// Strings, written with double quotes, are the same from the perspective of equality.
+// However, they are always different from the perspective of identity:
+"we are all the same" == "we are all the same" // true
+"we are all the same" === "we are all the same" // false: same, but not identical
+
+// Symbols, written with single quotes or following \ without spaces, are special:
+// There is only one of each.
+// Two representations of the same symbol are identical:
+'we are all the same' == 'we are all the same' // true
+'we are all the same' === 'we are all the same' // true
+::
+
+note::
+The relationship between link::#-equality:: and link::#-hash::, and link::#-identity:: and link::#-identityHash:::
+
+list::
+##link::Classes/Symbol:::
+
+The link::#-hash:: and link::#-identityHash:: methods return the same emphasis::hash table index:::
+
+list::
+##code::
+'we are all the same'.hash
+::
+teletype::
+-> 1592359842
+::
+
+##code::
+'we are all the same'.identityHash
+::
+teletype::
+-> 1592359842
+::
+
+##code::
+'we are all the same'.hash ==  'we are all the same'.identityHash
+::
+teletype::
+-> true
+::
+
+##code::
+'we are all the same'.hash === 'we are all the same'.identityHash
+::
+teletype::
+-> true
+::
+::
+
+##link::Classes/String::
+
+The link::#-hash:: and link::#-identityHash:: methods return the different emphasis::hash table index:::
+The link::#-hash:: method returns always the same emphasis::hash table index::, while link::#-identityHash:: method returns always the same emphasis::hash table index:::
+
+list::
+##code::
+"we are all the same".hash
+::
+returns always the following result:
+teletype::
+-> 1592359842
+::
+
+##code::
+"we are all the same".identityHash
+::
+returns new values each time it is evaluated:
+
+for example:
+teletype::
+-> -527806384
+::
+teletype::
+-> 1088233360
+::
+
+##code::
+"we are all the same".hash ==  "we are all the same".identityHash
+::
+teletype::
+-> false
+::
+
+##code::
+"we are all the same".hash === "we are all the same".identityHash
+::
+teletype::
+-> false
+::
+::
+::
+::
+
+
+
```

Comment:
I don't think you should talk about hash or identity hash here. It is more of a consequence of equality and identity and not necessary here.

Reply:
I think the conversation about hash could be move somewhere closer to dictionary or event, as this is where it really matters.

Reply:
How about moving to the new file as @telephon suggests?

Reply:
Julian suggested a new file to talk about equality and identity. 

What I'm saying is that hashing is a side effect and not directly relevant. In fact, it's pretty rare to use hashing directly.

The implications for event and dictionary are important, but best discussed in those help files, and linked to in specific classes.